velox/functions/prestosql/tests/ArraySortTest.cpp

```diff
@@ -431,6 +431,32 @@ TEST_P(ArraySortTest, dictionaryEncodedElements) {
           {{4, std::nullopt}, {1, std::nullopt, std::nullopt}}));
 }
 
+TEST_F(ArraySortTest, complexArrayDecode) {
+  // Tests to see whether we handle decoding in cases array is composed of
+  // nested complex types.
+  auto arrayVector =
+      makeArrayVector<int64_t>({{1, 2}, {0, -1}, {3, 4}, {5, 6}});
+  auto dictionaryVector = BaseVector::wrapInDictionary(
+      makeNulls(8, nullEvery(3)),
+      makeIndices(8, [](auto row) { return (row + 1) % 4; }),
+      8,
+      arrayVector);
+  auto arrayOfArrays = makeArrayVector({0, 2, 4, 6}, dictionaryVector);
+  auto result = evaluate("array_sort(c0)", makeRowVector({arrayOfArrays}));
```

Comment:
```
nit: s/result/sortedResult/
```

Reply:
We use result in every other so sortedResult is inconsistent I feel though  I can make the change if you feel strongly enough.